src/module.c

```diff
@@ -7646,6 +7646,8 @@ void moduleGILBeforeUnlock() {
     if (server.busy_module_yield_flags) {
         blockingOperationEnds();
         server.busy_module_yield_flags = BUSY_MODULE_YIELD_NONE;
+        if (server.current_client)
```

Comment:
here `server.current_client` is always NULL I think.

Reply:
yes, i thought so too, but then was worried maybe someone's using these contexts inside the redis's main thread while executing a command.
@MeirShpilraien @guybe7 please comment.

Reply:
I think that unlocking the GIL on the Redis main thread is a miss-use of the API...

Reply:
so shall i remove the protect / unprotect calls from here?
maybe add an assertion for currecnt_client being NULL?

Reply:
i guess i'd rather leave it there even if it might be dead code..